Composer/packages/adaptive-form/src/components/fields/OneOfField/OneOfField.tsx

```diff
@@ -38,36 +104,89 @@ const OneOfField: React.FC<FieldProps> = (props) => {
     }
   };
 
-  const Field = resolveFieldWidget(selectedSchema || {}, props.uiOptions, formUIOptions);
+  const renderField = () => {
+    if (!selectedSchema || Array.isArray(selectedSchema.type) || !selectedSchema.type) {
+      return null;
+    }
+    // attempt to get a placeholder with the selected schema
+    const placeholder = getUiPlaceholder({ ...props, schema: selectedSchema }) || props.placeholder;
+    const enumOptions = selectedSchema?.enum as string[];
+    const expression = schema?.oneOf?.some(({ $role }: any) => $role === 'expression');
+
+    const { field: Field, customProps } = resolveFieldWidget({
+      schema: selectedSchema,
+      uiOptions,
+      globalUIOptions: formUIOptions,
+      value,
+      expression: props.expression,
+      isOneOf: options.length > 1,
+    });
+    return (
+      <Field
+        key={selectedSchema.type}
+        expression={expression}
+        {...props}
+        {...customProps}
+        css={{ label: 'ExpressionFieldValue' }}
+        enumOptions={enumOptions}
+        // allow object fields to render their labels
+        label={selectedSchema.type !== 'object' ? false : undefined}
+        placeholder={placeholder}
+        schema={selectedSchema}
+        transparentBorder={false}
+        uiOptions={{ ...props.uiOptions, helpLink: 'https://bing.com' }}
```

Comment:
what's up with bing.com :D

Reply:
It is critical to use the secure protocol to go to bing too!